### ISSUE TITLE
Error in chromosome X 

### ISSUE THREAD

**Issue by xingejun** [saigegit/SAIGE#32]
Hi, I am using SAIGE and SAIGE-GENE for analysis with 22 autosome chromosome and chromosome X .

At step1, autosome chromosome goes well but chromosome X goes with the error both in SAIGE and SAIGE-GENE:

`Error in if (genoInd == 0) { : missing value where TRUE/FALSE needed
 Calls: fitNULLGLMM -> [extractVarianceRatio]`

bed, bim, fam file was used in my analysis, I have checked the log ,it seems like there are NA  in genoInd, but I don't know how it happens. I have post my step1 shell file and the log file.

Looking forward to your reply. 

[step1.txt](https://github.com/saigegit/SAIGE/files/8739052/step1.txt)
[step1.x.o284667.txt](https://github.com/saigegit/SAIGE/files/8739055/step1.x.o284667.txt)

**Comment by saigegit** [comment]
Hi @xingejun, 

In Step 1, the genetic relationship matrix (GRM) is constructed only with autosomal markers. 

Actually, in Step 1, the plink file should contain markers for constructing GRM (estimating sample relatedness coefficients) instead of all markers that you want to test the associations. Ideally, the number of markers is not smaller than the number of samples. 
For imputation data, in Step 1, we usually use the LD-pruned hard-called markers. 

In Step 2, all input markers will be tested for associations. The number of markers is usually much larger than the number of markers used for GRM in Step 1.  

Please see the examples here. 
https://saigegit.github.io//SAIGE-doc/docs/single_example.html#example-1

Thanks,
Wei

**Comment by xingejun** [comment]
Hi, Wei!

I have seen the example, but I have not figure out it. 

It means that I should do the assosiation test for chromosome X without Step1, just Step2, is it?

And In my case, the LD-pruned hard-called markers were used at Step 1 and they were imputed. But accutually, the makers are the same both in Step1 and Step2. A full GRM was constructed with all these markers. Considering your above suggestions, I am confusing if it is right? 

Thanks again!

**Comment by xingejun** [comment]
> Hi @xingejun,
> 
> In Step 1, the genetic relationship matrix (GRM) is constructed only with autosomal markers.
> 
> Actually, in Step 1, the plink file should contain markers for constructing GRM (estimating sample relatedness coefficients) instead of all markers that you want to test the associations. Ideally, the number of markers is not smaller than the number of samples. For imputation data, in Step 1, we usually use the LD-pruned hard-called markers.
> 
> In Step 2, all input markers will be tested for associations. The number of markers is usually much larger than the number of markers used for GRM in Step 1.
> 
> Please see the examples here. https://saigegit.github.io//SAIGE-doc/docs/single_example.html#example-1
> 
> Thanks, Wei

Hi, Wei!

I have seen the example, but I have not figure out it.

It means that I should do the assosiation test for chromosome X without Step1, just Step2, is it?

And In my case, the LD-pruned hard-called markers were used at Step 1 and they were imputed. But accutually, the makers are the same both in Step1 and Step2. A full GRM was constructed with all these markers. Considering your above suggestions, I am confusing if it is right?

Thanks again!

**Comment by xingejun** [comment]
Thanks!

xingejun

**Comment by xingejun** [comment]
Hi Wei!

Given your suggestions, I am wondering if it is possible for me to randomly select some makers to construct the full genetic relationship matrix (GRM)  as your example [https://saigegit.github.io//SAIGE-doc/docs/single_step1.html](https://saigegit.github.io//SAIGE-doc/docs/single_step1.html) But in the example, it seems like the selection only used when a sparse GRM is used to fit the null model.

Looking forward to your reply, again.

Thanks,
xin

**Comment by xingejun** [comment]
Hi Wei.

I have met another issue.

At Step 2 for SAIGE-GENE, I am wondering if i can use  the same one bfile with 22 chromosomes merged together, with --chrom  used for special chromosome. And that is to say,I don't know if it can go well when the --plinkFile(22 chromosomes in it) did not match the --groupFile(only 1 chromosome information in it) with the option --chromosome used. And I haven't found similiar examples at examples.

Thanks again!
xin

**Comment by xingejun** [comment]
Hi Wei,

I have tried use the same one bfile with 22 autosomal makers for step1 and step2, and it seems going well. 

However, when I try to fit the null model for chromosome X using the sparse GRM constructed with the 22 autosomal markers, there is a error again:

`missing value where TRUE/FALSE needed Calls: fitNULLGLMM -> [extractVarianceRatio]
Calls: fitNULLGLMM -> extractVarianceRatio`

So, I checked the --help for step1, and I added --includeNonautoMarkersforVarRatio=TRUE to my code. Now everything goes well, but I am not sure if it is right.

Thanks again，
xin
